frontend/src/pages/hardwareProfiles/nodeResource/ManageNodeResourceModal.tsx

```diff
@@ -0,0 +1,69 @@
+import React, { useEffect, useState } from 'react';
+import { Modal } from '@patternfly/react-core/deprecated';
+import DashboardModalFooter from '~/concepts/dashboard/DashboardModalFooter';
+import { Identifier } from '~/types';
+import { EMPTY_IDENTIFIER } from './const';
+import NodeResourceForm from './NodeResourceForm';
+import { validateDefaultCount, validateMinCount } from './utils';
+
+type ManageNodeResourceModalProps = {
+  onClose: () => void;
+  existingIdentifier?: Identifier;
+  onSave: (identifier: Identifier) => void;
+};
+
+const ManageNodeResourceModal: React.FC<ManageNodeResourceModalProps> = ({
+  onClose,
+  existingIdentifier,
+  onSave,
+}) => {
+  const [identifier, setIdentifier] = useState<Identifier>(EMPTY_IDENTIFIER);
+
+  const isButtonDisabled =
+    !identifier.displayName ||
+    !identifier.identifier ||
+    !validateDefaultCount(identifier) ||
+    !validateMinCount(identifier);
+
+  useEffect(() => {
+    if (existingIdentifier) {
+      setIdentifier(existingIdentifier);
+    }
+  }, [existingIdentifier]);
```

Comment:
you dont need to do this. instead update your useState to 
```tsx
const [identifier, setIdentifier] = useState<Identifier>(
    existingIdentifier || EMPTY_IDENTIFIER
  );
```
This is becuase you are unmounting the whole modal so you can use the default. im pretty sure